### PR TITLE
Update admin config to display more info & make searchable (Issue 4)

### PR DESCRIPTION
**Pull request by LauraMcCluskey7**

### CODE REVIEW

**Review by erikwaskiewicz — COMMENTED** [review]
Looks good, I've added a few more fields and search bars, can you review the code that I've written please?

The changes from issue 15 are appearing in this pull request too, I'm not sure why, but I'll check that they are correct and that we're not reverting those changes

**Comment by erikwaskiewicz** [comment]
The files from issue 15 are still correct, only `models.py` and `admin.py` need a review

File: analysis/admin.py [hunk 1]
```diff
@@ -2,85 +2,167 @@
...
     list_display = ('id', 'genomic_37', 'genomic_38')
+    search_fields = ['id', 'genomic_37', 'genomic_38']
+
 admin.site.register(Variant, VariantAdmin)
 
 
 class VariantInstanceAdmin(admin.ModelAdmin):
-    list_display = ('id', 'sample', 'variant')
+    list_display = ('id', 'get_sample', 'get_var', 'gene', 'hgvs_c', 'hgvs_p', 'vaf', 'final_decision')
+    search_fields = ['id', 'sample__sample_id', 'variant__genomic_37', 'gene', 'hgvs_c', 'hgvs_p', 'vaf', 'final_decision']
```
Comment (LauraMcCluskey7) [inline analysis/admin.py:69]: Search by final_decision doesn't work

File: analysis/admin.py [hunk 2]
```diff
@@ -2,85 +2,167 @@
...
 class VariantPanelAnalysisAdmin(admin.ModelAdmin):
     list_display = ('id', 'sample_analysis', 'variant_instance')
+    search_fields = ['id', 'sample_analysis__id', 'variant_instance__id']
+
 admin.site.register(VariantPanelAnalysis, VariantPanelAnalysisAdmin)
 
+
+class VariantCheckAdmin(admin.ModelAdmin):
+    list_display = ('id', 'variant_analysis', 'check_object', 'decision')
+    search_fields = ['id', 'variant_analysis__id', 'check_object__id', 'decision']
```
Comment (LauraMcCluskey7) [inline analysis/admin.py:95]: Search by decision doesn't work

File: analysis/admin.py [hunk 3]
```diff
@@ -2,85 +2,167 @@
...
+
+admin.site.register(VariantCheck, VariantCheckAdmin)
+
+
 class VariantListAdmin(admin.ModelAdmin):
-    list_display = ( 'name', 'list_type' )
+    list_display = ('name', 'list_type')
+    search_fields = ['name', 'list_type']
```
Comment (LauraMcCluskey7) [inline analysis/admin.py:102]: search by list_type doesn't work

File: analysis/admin.py [hunk 4]
```diff
@@ -2,85 +2,167 @@
...
+    list_display = ('id', 'fusion_genes', 'left_breakpoint', 'right_breakpoint')
+    search_fields = ['id', 'fusion_genes', 'left_breakpoint', 'right_breakpoint']
+
+admin.site.register(Fusion, FusionAdmin)
 
 
 class FusionAnalysisAdmin(admin.ModelAdmin):
-    list_display = ( 'sample', 'fusion_genes' )
+    list_display = ('id', 'sample', 'fusion_genes', 'hgvs', 'fusion_supporting_reads', 'ref_reads_1', 'ref_reads_2', 'final_decision')
+    search_fields = ['id', 'sample__id', 'fusion_genes__id', 'hgvs', 'fusion_supporting_reads', 'ref_reads_1', 'ref_reads_2', 'final_decision']
```
Comment (LauraMcCluskey7) [inline analysis/admin.py:152]: Cannot search by final_decision

**Review by LauraMcCluskey7 — COMMENTED** [review]
Small changes needed (see comments). 

The formatting of the admin page has also changed- could be due to updating the django version?

File: analysis/admin.py [hunk 5]
```diff
@@ -2,85 +2,167 @@
...
+
 admin.site.register(FusionAnalysis, FusionAnalysisAdmin)
 
 
 class FusionCheckAdmin(admin.ModelAdmin):
-    list_display = ( 'fusion_analysis', 'check_object' , 'decision')
+    list_display = ('id', 'fusion_analysis', 'check_object', 'decision')
+    search_fields = ['id', 'fusion_analysis__id', 'check_object__id', 'decision']
```
[the remark this answers is not in the document]
Comment (erikwaskiewicz) [reply analysis/admin.py]: Removed in commit ccef21b, wasn't searching due to the choices lookup dictionary, unsure on how to make it work with these and not critical to add this

File: analysis/admin.py (hunk 1 above)
Comment (erikwaskiewicz) [reply analysis/admin.py]: Removed in commit ccef21b, wasn't searching due to the choices lookup dictionary, unsure on how to make it work with these and not critical to add this

File: analysis/admin.py (hunk 2 above)
Comment (erikwaskiewicz) [reply analysis/admin.py]: Removed in commit ccef21b, wasn't searching due to the choices lookup dictionary, unsure on how to make it work with these and not critical to add this

File: analysis/admin.py (hunk 4 above)
Comment (erikwaskiewicz) [reply analysis/admin.py]: Removed in commit ccef21b, wasn't searching due to the choices lookup dictionary, unsure on how to make it work with these and not critical to add this

File: analysis/admin.py (hunk 3 above)
Comment (erikwaskiewicz) [reply analysis/admin.py]: Removed in commit ccef21b, wasn't searching due to the choices lookup dictionary, unsure on how to make it work with these and not critical to add this

File: analysis/admin.py [hunk 6]
```diff
@@ -2,85 +2,167 @@
...
+        return obj.panel.panel_name
+    get_panel.short_description = 'Panel'
+    get_panel.admin_order_field = 'panel__panel_name'
+
 admin.site.register(SampleAnalysis, SampleAnalysisAdmin)
 
+
 class CheckAdmin(admin.ModelAdmin):
     list_display = ('id', 'analysis', 'status', 'user')
+    search_fields = ['id', 'analysis__id', 'status', 'user__username']
```
[the remark this answers is not in the document]
Comment (erikwaskiewicz) [reply analysis/admin.py]: Removed in commit ccef21b, wasn't searching due to the choices lookup dictionary, unsure on how to make it work with these and not critical to add this

**Comment by erikwaskiewicz** [comment]
Changes from other PRs have been pulled into this one accidentally, only files to have changed are `admin.py` and `README.md`

Issues RE: JSONField were fixed in PR #24, it was due to the version of the auditlog package, updating Django also fixed the issue but caused the weird formatting (we've seen this in other databases)

I've removed the fields where the search wasn't working, this was due to the choices dictionary in the models, the indexes were being searched for instead of the full test (e.g. P instead of Pending), however P would also bring up anything with the letter P so it wasn't worth keeping. There is probably a way to fix this but a I couldn't find it in a quick Google search and it's only a nice to have feature, there's comments in the code in case we want to add in future

**Pull request merged**